src/pages/ProductDetail/ProductDetail.js

```diff
@@ -0,0 +1,178 @@
+import React, { Component } from 'react';
+import { GET_PRODUCTS_API, POST_ADD_CART_API } from '../../config.js';
+import { FaCreativeCommons, FaCreativeCommonsPdAlt } from 'react-icons/fa';
+import { RiArrowDownSLine, RiArrowUpSLine } from 'react-icons/ri';
+
+import './ProductDetail.scss';
+
+export class ProductDetail extends Component {
+  constructor() {
+    super();
+    this.set = React.createRef();
+    this.state = {
+      productData: {},
+      cartToggleOn: false,
+      recommendToggleOn: false,
+      addedCartAlertList: [],
+      backgroundColor: {
+        1: '#E9F9FE',
+        2: '#E0B5BA',
+        3: '#EFE9D9',
+        4: '#B1D9F1',
+        5: '#EAE9E9',
+        6: '#AEE19E',
+        7: '#FFF0A6',
+        8: '#CBC5E8',
+        9: '#FAD4BF',
+        10: '#E9F9FE',
+      },
+    };
+  }
+
+  componentDidMount() {
+    fetch(`${GET_PRODUCTS_API}/1.json`)
+      .then(res => res.json())
+      .then(data => {
+        this.setState({
+          productData: data,
+          cartToggleOn: data.cart_exist,
+        });
+      });
+  }
+
+  addedCartAlert = () => {
+    const { addedCartAlertList } = this.state;
+    addedCartAlertList.push('');
+    setTimeout(() => {
+      addedCartAlertList.pop();
+      this.setState({ addedCartAlertList: this.state.addedCartAlertList });
+    }, 3000);
+  };
+
+  postCart = () => {
+    const { cartToggleOn } = this.state;
+    if (cartToggleOn === true) {
+      this.addedCartAlert();
+    } else {
+      fetch(`${POST_ADD_CART_API}`, {
+        method: 'POST',
+        body: JSON.stringify({
+          productID: 1,
+        }),
+      });
+    }
+  };
+
+  addCart = () => {
+    this.postCart();
+    this.setState({ cartToggleOn: true });
+  };
+
+  toggleFunction = event => {
+    const name = event.currentTarget.name;
+    this.setState({ [`${name}ToggleOn`]: !this.state[`${name}ToggleOn`] });
+  };
+
+  render() {
+    if (!this.state.productData.productID) {
+      return <div>..Loading</div>;
+    } else {
+      const {
+        backgroundColor,
+        cartToggleOn,
+        recommendToggleOn,
+        addedCartAlertList,
+      } = this.state;
+      const {
+        icon_image_url,
+        productDescription,
+        productID,
+        productName,
+        productPrice,
+        thumbnail_image_url,
+        productTablet,
+      } = this.state.productData;
```

Comment:
```suggestion
    if (!this.state.productData.productID) {
      return <div>..Loading</div>;
    }
    
    const {
        backgroundColor,
        cartToggleOn,
        recommendToggleOn,
        addedCartAlertList,
      } = this.state;
    const {
        icon_image_url,
        productDescription,
        productID,
        productName,
        productPrice,
        thumbnail_image_url,
        productTablet,
      } = this.state.productData;
```
함수는 return을 만나면 종료됩니다.
render도 함수이므로 return을 만나면 종료되기 때문에 if문에서 early return 할 경우 else를 굳이 명시안해도 됩니다!